### PR TITLE
fix m1

### PR DESCRIPTION
**Pull request by reymondzzzz**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -53,16 +53,16 @@ tree-sitter-python = "0.20"
...
 async_once= "0.2.6"
 async-process = "2.0.1"
 itertools = "0.11.0"
 fst = { version = "0.4.7", features = ["levenshtein"] }
-lance = "=0.9.0"
-lance-linalg = "=0.9.0"
-lance-index = "=0.9.0"
+lance = { git = "https://github.com/lancedb/lance.git", branch = "main"}
+lance-linalg = { git = "https://github.com/lancedb/lance.git", branch = "main"}
+lance-index = { git = "https://github.com/lancedb/lance.git", branch = "main"}
```
Comment (valaises) [inline Cargo.toml:65]: we should remove it as soon as lance updates its crates

File: src/vecdb/handler.rs [hunk 2]
```diff
@@ -641,19 +635,20 @@ impl VecDBHandler {
...
         embedding: Vec<f32>,
-        top_n: usize
-    ) -> vectordb::error::Result<Vec<Record>> {
+        top_n: usize,
+    ) -> lancedb::error::Result<Vec<Record>> {
         let query = self
             .data_table
             .clone()
-            .search(Some(Float32Array::from(embedding.clone())))
+            .search(&*embedding.clone())
```
Comment (valaises) [inline src/vecdb/handler.rs:643]: Scary

**Review by valaises — APPROVED** [review]

**Review by valaises — CHANGES_REQUESTED** [review]
I don't like unwrap

**Comment by valaises** [comment]
image
Why do you need `&*` here?

**Review by JegernOUTT — APPROVED** [review]

**Pull request closed without merging**